### PR TITLE
#AIR-909 Security fixes in kubernetes dashboard v2.4.0

### PR DESCRIPTION
**Pull request by ShaunakJoshi1407**
Running docker build locally using npm:

```
npm run docker:build                                                                

> kubernetes-dashboard@2.4.0 docker:build /Users/shaunakj/go/src/github.com/platform9/kuberenetesui-dashboard
> npm run clean && npm run build:cross && gulp docker-image:release:cross


> kubernetes-dashboard@2.4.0 clean /Users/shaunakj/go/src/github.com/platform9/kuberenetesui-dashboard
> rm -rf .go_workspace .tmp coverage dist npm-debug.log


> kubernetes-dashboard@2.4.0 build:cross /Users/shaunakj/go/src/github.com/platform9/kuberenetesui-dashboard
> ./aio/scripts/build.sh -c


> kubernetes-dashboard@2.4.0 postversion /Users/shaunakj/go/src/github.com/platform9/kuberenetesui-dashboard
> node aio/scripts/version.mjs

Version v2.4.0-2-gdd2911c47 saved to kuberenetesui-dashboard/src/app/frontend/environments/version.ts

Building backends for all supported architectures
[13:35:18] Requiring external module @babel/register
[13:35:18] Using gulpfile ~/go/src/github.com/platform9/kuberenetesui-dashboard/gulpfile.babel.js
[13:35:18] Starting 'backend:prod:cross'...
[13:35:18] Starting '<anonymous>'...
[13:39:45] Finished '<anonymous>' after 4.43 min
[13:39:45] Finished 'backend:prod:cross' after 4.43 min

Building localized frontend
✔ Browser application bundle generation complete.
✔ Localized bundle generation complete.
✔ Copying assets complete.
✔ Index html generation complete.

Initial Chunk Files               | Names         |      Size
main.876324fb5cfd90f85cfc.js      | main          |   2.57 MB
styles.aa0538c9a91ebbb04705.css   | styles        | 230.72 kB
polyfills.2565916e4afd13edaa84.js | polyfills     | 108.06 kB
scripts.f76573725d49abb057d3.js   | scripts       |  69.31 kB
runtime.49389a706ff331d65932.js   | runtime       |   3.92 kB

                                  | Initial Total |   2.98 MB

Lazy Chunk Files                  | Names         |      Size
719.65caf4b1a1d47bc70e2a.js       | -             | 336.97 kB
288.6e53d304ea1ba1e8ee8d.js       | -             |  76.79 kB
956.53eb258658a0f4f515bf.js       | -             |  45.80 kB
636.c848a25ebb083078ba3a.js       | -             |  32.42 kB
356.010b40ce6e73f3e641bb.js       | -             |  25.37 kB
987.df15677429fc826a2e0b.js       | -             |  25.08 kB
782.71921965ef8756d2e1e2.js       | -             |  17.79 kB
876.a15a56f74c7baa412814.js       | -             |  13.27 kB
817.eb6362f31405fbe77dc6.js       | -             |   9.12 kB
19.166b84ad6f2ebd0ef430.js        | -             |   8.74 kB
161.a4311ca14943212a7a70.js       | -             |   8.50 kB
538.734b24d210095ee18d4b.js       | -             |   8.48 kB
355.fe92d532bbf05c4308c6.js       | -             |   7.90 kB
117.6b0e667e9ccd0fb6c515.js       | -             |   7.61 kB
common.b5c0dffe3aed7315bdcc.js    | common        |   7.03 kB
783.a909f7162e8d76aadede.js       | -             |   6.81 kB
45.c5f0fa13f7c1bf2510a3.js        | -             |   6.51 kB
43.6f97c1062140c72d0e2d.js        | -             |   6.20 kB
338.d3bf40a220d25225493e.js       | -             |   5.95 kB
163.f65f05d51be364d8d527.js       | -             |   5.93 kB
437.b6f71306de2f7805f7c2.js       | -             |   5.79 kB
760.e533845767b2db410d65.js       | -             |   5.10 kB
292.81c997c3e38e002fb78d.js       | -             |   4.70 kB
135.99afb2ddd3929b595bc3.js       | -             |   4.59 kB
959.d2b696eef6747bd5bb90.js       | -             |   4.54 kB
300.6978bb628fcc8c7b93e1.js       | -             |   3.70 kB
40.19d45f1f38c81e0648b3.js        | -             |   3.04 kB
213.c3559e89e5550f91f53b.js       | -             |   3.02 kB
673.6f2d546a5517926770af.js       | -             |   2.57 kB
594.a8b1aba0e9bc31dd2514.js       | -             |   2.54 kB
859.1d2b3a44394e7e1fd4d3.js       | -             |   2.31 kB
627.c3b83af4375f2c116b0b.js       | -             | 998 bytes
901.053f4c19e1f6f30615e4.js       | -             | 416 bytes
439.01bd554088d5b76c1f07.js       | -             | 405 bytes
375.340b8d1daeb2aa66404e.js       | -             | 404 bytes
671.3f6c9df1792f12f553ae.js       | -             | 404 bytes
699.56445c8cf7554745035d.js       | -             | 402 bytes
954.c5869cd7b30bccff8517.js       | -             | 401 bytes
21.61cbcb5bb0835bbcedaa.js        | -             | 396 bytes

Build at: 2023-04-18T08:11:51.821Z - Hash: a97fcfe49afd6b2165a5 - Time: 113088ms

Warning: /Users/shaunakj/go/src/github.com/platform9/kuberenetesui-dashboard/node_modules/@swimlane/ngx-charts/fesm2015/swimlane-ngx-charts.js depends on 'clone-deep'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



Copying frontend to backend dist dir

Copying locales file to backend dist dirs

Copying Dockerfile to backend dist dirs

Build finished successfully after 409s
[13:42:07] Requiring external module @babel/register
[13:42:07] Using gulpfile ~/go/src/github.com/platform9/kuberenetesui-dashboard/gulpfile.babel.js
[13:42:07] Starting 'docker-image:release:cross'...
[13:42:07] Starting 'docker-file:cross'...
[13:42:07] Finished 'docker-file:cross' after 44 ms
[13:42:07] Starting '<anonymous>'...
Sending build context to Docker daemon  223.7MB
Step 1/8 : FROM ubuntu:20.04 as builder 219.5MB
 ---> 1c5c8d0b973a
Step 2/8 : RUN useradd -u 10001 nonroot
 ---> Using cachetext to Docker daemon  215.6MB
 ---> 3375f45d8c8d
Step 3/8 : FROM scratch as final
 ---> 
Step 4/8 : ADD . /
Sending build context to Docker daemon    225MB
Step 1/8 : FROM ubuntu:20.04 as builder
 ---> 1c5c8d0b973a
Step 2/8 : RUN useradd -u 10001 nonroot
 ---> Using cache
 ---> 3375f45d8c8d
Step 3/8 : FROM scratch as final
 ---> 
Step 4/8 : ADD . /
Sending build context to Docker daemon  221.9MB
Step 1/8 : FROM ubuntu:20.04 as builder
 ---> 1c5c8d0b973a
Step 2/8 : RUN useradd -u 10001 nonroot
 ---> Using cache
 ---> 3375f45d8c8d
Step 3/8 : FROM scratch as final
 ---> 
Step 4/8 : ADD . /
Sending build context to Docker daemon  227.2MB
Step 1/8 : FROM ubuntu:20.04 as builder
 ---> 1c5c8d0b973a
Step 2/8 : RUN useradd -u 10001 nonroot
 ---> Using cache
 ---> 3375f45d8c8d
Step 3/8 : FROM scratch as final
 ---> 
Step 4/8 : ADD . /
Sending build context to Docker daemon  222.8MB
Step 1/8 : FROM ubuntu:20.04 as builder
 ---> 1c5c8d0b973a
Step 2/8 : RUN useradd -u 10001 nonroot
 ---> Using cache
 ---> 3375f45d8c8d
Step 3/8 : FROM scratch as final
 ---> 
Step 4/8 : ADD . /
 ---> 833f44fa5fd4
Step 5/8 : COPY --from=builder /etc/passwd /etc/passwd
 ---> f9deb57b5fb9
Step 5/8 : COPY --from=builder /etc/passwd /etc/passwd
 ---> fc8af83a7820
Step 5/8 : COPY --from=builder /etc/passwd /etc/passwd
 ---> 6573dac75fa9
Step 5/8 : COPY --from=builder /etc/passwd /etc/passwd
 ---> b0fb04f2dc88
Step 5/8 : COPY --from=builder /etc/passwd /etc/passwd
 ---> 4f78849db064
Step 6/8 : USER nonroot
 ---> 96926ae67d53
Step 6/8 : USER nonroot
 ---> d70ecf2fb953
Step 6/8 : USER nonroot
 ---> c31a33731975
Step 6/8 : USER nonroot
 ---> 9a50483c49e9
Step 6/8 : USER nonroot
 ---> Running in 1cb9ce51fd0b
 ---> Running in 5005397d5da8
 ---> Running in 26c8eec3af7b
 ---> Running in bb665ce0a768
 ---> Running in 83b697f68c44
Removing intermediate container 1cb9ce51fd0b
 ---> de9c55d8848e
Step 7/8 : EXPOSE 9090 8443
Removing intermediate container 5005397d5da8
 ---> 4eec2d655270
Step 7/8 : EXPOSE 9090 8443
Removing intermediate container bb665ce0a768
Removing intermediate container 26c8eec3af7b
 ---> 31d170af7377
Step 7/8 : EXPOSE 9090 8443
 ---> 12aa1cb46a6f
Step 7/8 : EXPOSE 9090 8443
Removing intermediate container 83b697f68c44
 ---> 220a47078e2b
Step 7/8 : EXPOSE 9090 8443
 ---> Running in 04f31c6de9ed
 ---> Running in e90fa0261dd0
 ---> Running in db7f2df7a80a
 ---> Running in 4cfe2b8258c5
 ---> Running in d9eb7775ba71
Removing intermediate container 04f31c6de9ed
 ---> 2ed91c1e1cc7
Step 8/8 : ENTRYPOINT ["/dashboard", "--insecure-bind-address=0.0.0.0", "--bind-address=0.0.0.0"]
Removing intermediate container db7f2df7a80a
 ---> 5116c4a09358
Step 8/8 : ENTRYPOINT ["/dashboard", "--insecure-bind-address=0.0.0.0", "--bind-address=0.0.0.0"]
Removing intermediate container e90fa0261dd0
 ---> d1e679976b75
Step 8/8 : ENTRYPOINT ["/dashboard", "--insecure-bind-address=0.0.0.0", "--bind-address=0.0.0.0"]
Removing intermediate container d9eb7775ba71
 ---> 7560b6c51ccb
Step 8/8 : ENTRYPOINT ["/dashboard", "--insecure-bind-address=0.0.0.0", "--bind-address=0.0.0.0"]
Removing intermediate container 4cfe2b8258c5
 ---> e807a37c2381
Step 8/8 : ENTRYPOINT ["/dashboard", "--insecure-bind-address=0.0.0.0", "--bind-address=0.0.0.0"]
 ---> Running in 0cb7475c2493
 ---> Running in bfe97ca3c7e8
 ---> Running in 27fb6a4614fe
 ---> Running in 7e3e94a83f7b
 ---> Running in 7e63886e0c8e
Removing intermediate container 27fb6a4614fe
 ---> e024baee9ed3
Removing intermediate container 0cb7475c2493
 ---> a4dcf8be115f
Removing intermediate container bfe97ca3c7e8
 ---> 03b5a07879b2
Removing intermediate container 7e63886e0c8e
Removing intermediate container 7e3e94a83f7b
 ---> b3a9b5214f5e
 ---> 9b22855ec263
Successfully built a4dcf8be115f
Successfully built e024baee9ed3
Successfully tagged kubernetesui/dashboard-s390x:v2.4.0-pmk
Successfully tagged kubernetesui/dashboard-ppc64le:v2.4.0-pmk
Successfully built 03b5a07879b2
Successfully tagged kubernetesui/dashboard-amd64:v2.4.0-pmk
Successfully built 9b22855ec263
Successfully tagged kubernetesui/dashboard-arm:v2.4.0-pmk
Successfully built b3a9b5214f5e
Successfully tagged kubernetesui/dashboard-arm64:v2.4.0-pmk

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[13:43:33] Finished '<anonymous>' after 1.42 min
[13:43:33] Finished 'docker-image:release:cross' after 1.42 min
```

Listing the docker images created:
```
docker images | grep kubernetesui/dashboard
kubernetesui/dashboard-ppc64le                         v2.4.0-pmk        c83e46dfd445   53 seconds ago   221MB
kubernetesui/dashboard-amd64                           v2.4.0-pmk        73d4eaf9e5e5   53 seconds ago   222MB
kubernetesui/dashboard-arm                             v2.4.0-pmk        a5499167bde4   55 seconds ago   219MB
kubernetesui/dashboard-arm64                           v2.4.0-pmk        816fc378f23d   55 seconds ago   220MB
kubernetesui/dashboard-s390x                           v2.4.0-pmk        2b6f0c7aa30f   55 seconds ago   224MB
```

Trivy scan on a docker image created above:
```
trivy image 73d4eaf9e5e5            
2023-04-18T18:55:20.925+0530	INFO	Need to update DB
2023-04-18T18:55:20.925+0530	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2023-04-18T18:55:20.925+0530	INFO	Downloading DB...
36.59 MiB / 36.59 MiB [--------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 5.10 MiB p/s 7.4s
2023-04-18T18:55:31.980+0530	INFO	Vulnerability scanning is enabled
2023-04-18T18:55:31.980+0530	INFO	Secret scanning is enabled
2023-04-18T18:55:31.980+0530	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-04-18T18:55:31.980+0530	INFO	Please see also https://aquasecurity.github.io/trivy/v0.37/docs/secret/scanning/#recommendation for faster secret detection
2023-04-18T18:56:22.969+0530	INFO	Number of language-specific files: 1
2023-04-18T18:56:22.970+0530	INFO	Detecting gobinary vulnerabilities...

dashboard (gobinary)

Total: 1 (UNKNOWN: 0, LOW: 1, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

┌────────────────────────────────┬─────────────────────┬──────────┬─────────────────────┬───────────────┬───────────────────────────────────────────────────┐
│            Library             │    Vulnerability    │ Severity │  Installed Version  │ Fixed Version │                       Title                       │
├────────────────────────────────┼─────────────────────┼──────────┼─────────────────────┼───────────────┼───────────────────────────────────────────────────┤
│ github.com/docker/distribution │ GHSA-qq97-vm5h-rrhg │ LOW      │ v2.7.1+incompatible │ 2.8.0         │ OCI Manifest Type Confusion Issue                 │
│                                │                     │          │                     │               │ https://github.com/advisories/GHSA-qq97-vm5h-rrhg │
└────────────────────────────────┴─────────────────────┴──────────┴─────────────────────┴───────────────┴───────────────────────────────────────────────────┘

```

Tagging the build and pushing to my dockerhub:

```
$ docker image tag 73d4eaf9e5e5 shaunakjoshi/lts

$ docker image ls  
REPOSITORY                                             TAG               IMAGE ID       CREATED         SIZE
kubernetesui/dashboard-amd64                           v2.4.0-pmk        73d4eaf9e5e5   3 minutes ago   222MB
shaunakjoshi/lts                                       latest            73d4eaf9e5e5   3 minutes ago   222MB
kubernetesui/dashboard-ppc64le                         v2.4.0-pmk        c83e46dfd445   3 minutes ago   221MB
kubernetesui/dashboard-arm                             v2.4.0-pmk        a5499167bde4   3 minutes ago   219MB
kubernetesui/dashboard-s390x                           v2.4.0-pmk        2b6f0c7aa30f   3 minutes ago   224MB
kubernetesui/dashboard-arm64                           v2.4.0-pmk        816fc378f23d   3 minutes ago   220MB

$ docker push shaunakjoshi/lts:latest    
The push refers to repository [docker.io/shaunakjoshi/lts]
64a116ebbd5a: Pushed 
f4c9cbbb765f: Pushed 
latest: digest: sha256:325d13b84b4d1fc3e1ebe0c411b3eaa639b214ba0509e52ac3dba37a53a8aa53 size: 736
```

Editing the deployment on a 1.20 cluster and then checking the status:
```
$ kubectl describe deployment kubernetes-dashboard -n kubernetes-dashboard
Name:                   kubernetes-dashboard
Namespace:              kubernetes-dashboard
CreationTimestamp:      Tue, 18 Apr 2023 09:23:47 +0000
Labels:                 k8s-app=kubernetes-dashboard
Annotations:            deployment.kubernetes.io/revision: 3
Selector:               k8s-app=kubernetes-dashboard
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           k8s-app=kubernetes-dashboard
  Service Account:  kubernetes-dashboard
  Containers:
   kubernetes-dashboard:
    Image:      docker.io/shaunakjoshi/lts:latest
    Port:       8443/TCP
    Host Port:  0/TCP
    Args:
      --tls-cert-file=dashboard.crt
      --tls-key-file=dashboard.key
      --namespace=kubernetes-dashboard
    Liveness:     http-get https://:8443/ delay=30s timeout=30s period=10s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /certs from kubernetes-dashboard-certs (rw)
      /tmp from tmp-volume (rw)
  Volumes:
   kubernetes-dashboard-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  kubernetes-dashboard-certs
    Optional:    false
   tmp-volume:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Available      True    MinimumReplicasAvailable
  Progressing    True    NewReplicaSetAvailable
OldReplicaSets:  <none>
NewReplicaSet:   kubernetes-dashboard-68b6b56cdc (1/1 replicas created)
Events:
  Type    Reason             Age    From                   Message
  ----    ------             ----   ----                   -------
  Normal  ScalingReplicaSet  26m    deployment-controller  Scaled up replica set kubernetes-dashboard-b6f5569c4 to 1
  Normal  ScalingReplicaSet  3m4s   deployment-controller  Scaled down replica set kubernetes-dashboard-b6f5569c4 to 0
  Normal  ScalingReplicaSet  3m4s   deployment-controller  Scaled up replica set kubernetes-dashboard-68b6b56cdc to 1
  Normal  ScalingReplicaSet  2m53s  deployment-controller  Scaled down replica set kubernetes-dashboard-5674c696b9 to 0

$ kubectl get pods -A | grep dashboard
kubernetes-dashboard   dashboard-metrics-scraper-567d566849-2df49   1/1     Running            0          3h58m
kubernetes-dashboard   kubernetes-dashboard-68b6b56cdc-cqvgs        1/1     Running            0          13m

```

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -12,9 +12,4 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License
-. "$(dirname "$0")/_/husky.sh"
-
-npm run check
```
Comment (cruizen) [inline .husky/pre-push:17]: Can you just comment these out with an additional comment to explain why they are disabled?

**Review by cruizen — APPROVED** [review]

File: .husky/pre-push (hunk 1 above)
Comment (ShaunakJoshi1407) [reply .husky/pre-push:17]: Done

**Comment by ShaunakJoshi1407** [comment]
TC Build is working now: https://teamcity.platform9.horse/buildConfiguration/Pf9project_Platform9ComponentsForKubernetes_KuberenetesuiDashboard/2636348

Tested this image of kubernetes-dashboard on a 1.20 cluster and it is working fine :
```
$ kubectl describe deployment kubernetes-dashboard -n kubernetes-dashboard
Name:                   kubernetes-dashboard
Namespace:              kubernetes-dashboard
CreationTimestamp:      Wed, 19 Apr 2023 06:27:53 +0000
Labels:                 k8s-app=kubernetes-dashboard
Annotations:            deployment.kubernetes.io/revision: 4
Selector:               k8s-app=kubernetes-dashboard
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           k8s-app=kubernetes-dashboard
  Service Account:  kubernetes-dashboard
  Containers:
   kubernetes-dashboard:
    Image:      platform9/kubernetesui-dashboard:v2.4.0-pmk-2636348
    Port:       8443/TCP
    Host Port:  0/TCP
    Args:
      --tls-cert-file=dashboard.crt
      --tls-key-file=dashboard.key
      --namespace=kubernetes-dashboard
    Liveness:     http-get https://:8443/ delay=30s timeout=30s period=10s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /certs from kubernetes-dashboard-certs (rw)
      /tmp from tmp-volume (rw)
  Volumes:
   kubernetes-dashboard-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  kubernetes-dashboard-certs
    Optional:    false
   tmp-volume:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Available      True    MinimumReplicasAvailable
  Progressing    True    NewReplicaSetAvailable
OldReplicaSets:  <none>
NewReplicaSet:   kubernetes-dashboard-5d76d84cbf (1/1 replicas created)
Events:
  Type    Reason             Age                  From                   Message
  ----    ------             ----                 ----                   -------
  Normal  ScalingReplicaSet  3m32s                deployment-controller  Scaled up replica set kubernetes-dashboard-5674c696b9 to 1
  Normal  ScalingReplicaSet  3m30s                deployment-controller  Scaled down replica set kubernetes-dashboard-5d76d84cbf to 0
  Normal  ScalingReplicaSet  49s (x2 over 3m45s)  deployment-controller  Scaled up replica set kubernetes-dashboard-5d76d84cbf to 1
  Normal  ScalingReplicaSet  47s (x2 over 3m33s)  deployment-controller  Scaled down replica set kubernetes-dashboard-5674c696b9 to 0
  
$ kubectl get pods -A | grep kubernetes-dashboard
kubernetes-dashboard   dashboard-metrics-scraper-567d566849-4nx6f   1/1     Running            0          22h
kubernetes-dashboard   kubernetes-dashboard-5d76d84cbf-zgdnq        1/1     Running            0          2m18s
```

File: dist/amd64/dashboard
Comment (cruizen) [inline dist/amd64/dashboard:1]: Can we exclude this large (51 MB) binary file from the changeset?

File: dist/amd64/locale_conf.json
Comment (cruizen) [inline dist/amd64/locale_conf.json:1]: We didn't really have to include these translations in our fork - it's a big changeset and difficult to navigate (my browser struggled to add a comment here).

**Review by cruizen — CHANGES_REQUESTED** [review]

**Comment by cruizen** [comment]
> TC Build is working now: https://teamcity.platform9.horse/buildConfiguration/Pf9project_Platform9ComponentsForKubernetes_KuberenetesuiDashboard/2636348
> 
> Tested this image of kubernetes-dashboard on a 1.20 cluster and it is working fine :
> 
> ```
> $ kubectl describe deployment kubernetes-dashboard -n kubernetes-dashboard
> Name:                   kubernetes-dashboard
> Namespace:              kubernetes-dashboard
> CreationTimestamp:      Wed, 19 Apr 2023 06:27:53 +0000
> Labels:                 k8s-app=kubernetes-dashboard
> Annotations:            deployment.kubernetes.io/revision: 4
> Selector:               k8s-app=kubernetes-dashboard
> Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
> StrategyType:           RollingUpdate
> MinReadySeconds:        0
> RollingUpdateStrategy:  25% max unavailable, 25% max surge
> Pod Template:
>   Labels:           k8s-app=kubernetes-dashboard
>   Service Account:  kubernetes-dashboard
>   Containers:
>    kubernetes-dashboard:
>     Image:      platform9/kubernetesui-dashboard:v2.4.0-pmk-2636348
>     Port:       8443/TCP
>     Host Port:  0/TCP
>     Args:
>       --tls-cert-file=dashboard.crt
>       --tls-key-file=dashboard.key
>       --namespace=kubernetes-dashboard
>     Liveness:     http-get https://:8443/ delay=30s timeout=30s period=10s #success=1 #failure=3
>     Environment:  <none>
>     Mounts:
>       /certs from kubernetes-dashboard-certs (rw)
>       /tmp from tmp-volume (rw)
>   Volumes:
>    kubernetes-dashboard-certs:
>     Type:        Secret (a volume populated by a Secret)
>     SecretName:  kubernetes-dashboard-certs
>     Optional:    false
>    tmp-volume:
>     Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
>     Medium:     
>     SizeLimit:  <unset>
> Conditions:
>   Type           Status  Reason
>   ----           ------  ------
>   Available      True    MinimumReplicasAvailable
>   Progressing    True    NewReplicaSetAvailable
> OldReplicaSets:  <none>
> NewReplicaSet:   kubernetes-dashboard-5d76d84cbf (1/1 replicas created)
> Events:
>   Type    Reason             Age                  From                   Message
>   ----    ------             ----                 ----                   -------
>   Normal  ScalingReplicaSet  3m32s                deployment-controller  Scaled up replica set kubernetes-dashboard-5674c696b9 to 1
>   Normal  ScalingReplicaSet  3m30s                deployment-controller  Scaled down replica set kubernetes-dashboard-5d76d84cbf to 0
>   Normal  ScalingReplicaSet  49s (x2 over 3m45s)  deployment-controller  Scaled up replica set kubernetes-dashboard-5d76d84cbf to 1
>   Normal  ScalingReplicaSet  47s (x2 over 3m33s)  deployment-controller  Scaled down replica set kubernetes-dashboard-5674c696b9 to 0
>   
> $ kubectl get pods -A | grep kubernetes-dashboard
> kubernetes-dashboard   dashboard-metrics-scraper-567d566849-4nx6f   1/1     Running            0          22h
> kubernetes-dashboard   kubernetes-dashboard-5d76d84cbf-zgdnq        1/1     Running            0          2m18s
> ```

The build wasn't working yesterday - and I don't see code changes since?

**Pull request merged**